File: app/src/main/java/com/example/aichat/ProactiveMessageWorker.java
package com.example.aichat;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

public class ProactiveMessageWorker extends Worker {
    public ProactiveMessageWorker(@NonNull Context context, @NonNull WorkerParameters params) {
        super(context, params);
    }

    @NonNull
    @Override
    public Result doWork() {
        Context app = getApplicationContext();
        CountDownLatch latch = new CountDownLatch(1);
        new ProactiveMessageSyncManager(app).syncOnce(new ProactiveMessageSyncManager.SyncCallback() {
            @Override
            public void onComplete() {
                latch.countDown();
            }
        });
        try {
            latch.await(60, TimeUnit.SECONDS);
        } catch (InterruptedException ignored) {
            Thread.currentThread().interrupt();
        }
        ProactiveMessageWorkScheduler.scheduleNext(app);
        return Result.success();
    }
}

File: app/src/main/java/com/example/aichat/ProactiveMessageWorker.kt
package com.example.aichat

import android.content.Context
import androidx.annotation.NonNull
import androidx.work.Worker
import androidx.work.WorkerParameters
import java.util.concurrent.CountDownLatch
import java.util.concurrent.TimeUnit

class ProactiveMessageWorker(
    @NonNull context: Context,
    @NonNull params: WorkerParameters
) : Worker(context, params) {

    @NonNull
    override fun doWork(): Result {
        val app = applicationContext
        val latch = CountDownLatch(1)
        ProactiveMessageSyncManager(app).syncOnce(object : ProactiveMessageSyncManager.SyncCallback {
            override fun onComplete() {
                latch.countDown()
            }
        })
        try {
            latch.await(60, TimeUnit.SECONDS)
        } catch (ignored: InterruptedException) {
            Thread.currentThread().interrupt()
        }
        ProactiveMessageWorkScheduler.scheduleNext(app)
        return Result.success()
    }
}
